pandas/tests/test_categorical.py

```diff
@@ -3640,6 +3640,153 @@ def test_cat_accessor_no_new_attributes(self):
         with tm.assertRaisesRegexp(AttributeError, "You cannot add any new attribute"):
             c.cat.xlabel = "a"
 
+    def test_str_accessor_api_for_categorical(self):
+        # https://github.com/pydata/pandas/issues/10661
+        from pandas.core.strings import StringMethods
+        s = Series(list('aabb'))
+        s = s + " " + s
+        c = s.astype('category')
+        self.assertIsInstance(c.str, StringMethods)
+
+        # Handcrafted by
+        # s = pandas.Series(list("abcd"))
+        # t = "('%s', (), {})"
+        # functs = [t % f for f in dir(s.str) if not f.startswith("_")]
```

Comment:
so instead of listing these out as this would be instantly out-dated if something else is added.

create a list of the ones that need special arguments, then programatically create the rest